src/web_interface/rest/helper.py

```diff
@@ -0,0 +1,192 @@
+    try:
+        offset = int(request_parameters.get('offset', 0))
+    except (TypeError, ValueError):
+        raise ValueError('Malformed offset parameter')
+
+    try:
+        limit = int(request_parameters.get('limit', 0))
+    except (TypeError, ValueError):
+        raise ValueError('Malformed limit parameter')
+
+    return offset, limit
+
+
+def get_query(request_parameter: ImmutableMultiDict) -> dict:
+    '''
+    Parse the query parameter from request parameters. Query is a dictionary representing a MongoDB query.
+
+    :param request_parameters: dict containing the request parameters.
+    :return: The MongoDB query as dict.
+    '''
+    try:
+        query = request_parameter.get('query')
+        query = json.loads(query if query else '{}')
+    except (AttributeError, KeyError):
+        return dict()
+    except json.JSONDecodeError:
+        raise ValueError('Query must be a json document')
+    if not isinstance(query, dict):
+        raise ValueError('Query must be a json document')
+    return query if query else dict()
+
+
+def _get_boolean_from_request(request_parameters: ImmutableMultiDict, name: str) -> bool:
+    try:
+        parameter = json.loads(request_parameters.get(name, 'false'))
+        if not isinstance(parameter, bool):
+            raise TypeError()
+    except (AttributeError, KeyError):
+        return False
+    except (json.JSONDecodeError, TypeError):
+        raise ValueError('{} must be true or false'.format(name))
+    return parameter
+
+
+def get_tar_flag(request_parameters: ImmutableMultiDict) -> bool:
+    '''
+    Parse the tar flag from request parameters.
+
+    :param request_parameters: dict containing the request parameters.
+    :return: The flag as boolean.
+    '''
+    return _get_boolean_from_request(request_parameters, 'tar')
+
+
+def get_summary_flag(request_parameters: ImmutableMultiDict) -> bool:
+    '''
+    Parse the summary flag from request parameters.
+
+    :param request_parameters: dict containing the request parameters.
+    :return: The flag as boolean.
+    '''
+    return _get_boolean_from_request(request_parameters, 'summary')
+
+
+def get_recursive_flag(request_parameters: ImmutableMultiDict) -> bool:
+    '''
+    Parse the recursive flag from request parameters.
+
+    :param request_parameters: dict containing the request parameters.
+    :return: The flag as boolean.
+    '''
+    return _get_boolean_from_request(request_parameters, 'recursive')
+
+
+def get_inverted_flag(request_parameters: ImmutableMultiDict) -> bool:
+    '''
+    Parse the inverted flag from request parameters.
+
+    :param request_parameters: dict containing the request parameters.
+    :return: The flag as boolean.
+    '''
+    return _get_boolean_from_request(request_parameters, 'inverted')
+
+
+def get_update(request_parameters: ImmutableMultiDict) -> list:
+    '''
+    Parse the update parameter from request parameters. Update is a list of analysis plugins that shall be updated.
```

Comment:
Not to be nitpicky, but
```suggestion
    Parse the update parameter from request parameters. Update is a list of analysis plugins whose analysis results shall be updated.
```